### DISCUSSION TITLE
How do I get Index of a Button Inside MFXTable

### DISCUSSION THREAD

**Discussion by Newbie-12131** [palexdev/MaterialFX#316] [Q&A]
I have a table where each row contains a button and some information in different columns. I want to know how to make the button do a specific task based on the information present in the same row as the button. But I had no success in even getting the row Index.

and so even if I got the row Index on where the button is, how do I access items from the cells of another columns?

`
private void setMaintenanceTable() {

        // Define the columns for the table
        MFXTableColumn<MaintenanceTableClass> plateNumber = new MFXTableColumn<>("Plate Number", Comparator.comparing(MaintenanceTableClass::getPlateNumber));
        MFXTableColumn<MaintenanceTableClass> vehicleType = new MFXTableColumn<>("Vehicle Type", Comparator.comparing(MaintenanceTableClass::getVehicleType));
        MFXTableColumn<MaintenanceTableClass> maintenanceType = new MFXTableColumn<>("Maintenance Type", Comparator.comparing(MaintenanceTableClass::getMaintenanceType));
        MFXTableColumn<MaintenanceTableClass> vehicleModelTbl = new MFXTableColumn<>("Vehicle Model", Comparator.comparing(MaintenanceTableClass::getVehicleModel));
        MFXTableColumn<MaintenanceTableClass> maintenanceCheck = new MFXTableColumn<>("Maintenance Check", Comparator.comparing(MaintenanceTableClass::getMaintenanceCheck));
        MFXTableColumn<MaintenanceTableClass> returnButton = new MFXTableColumn<>("Action");

        // Set the row cell factories for the columns
        plateNumber.setRowCellFactory(maintenanceTableClass -> new MFXTableRowCell<>(MaintenanceTableClass::getPlateNumber));
        vehicleType.setRowCellFactory(maintenanceTableClass -> new MFXTableRowCell<>(MaintenanceTableClass::getVehicleType));
        maintenanceType.setRowCellFactory(maintenanceTableClass -> new MFXTableRowCell<>(MaintenanceTableClass::getMaintenanceType));
        vehicleModelTbl.setRowCellFactory(maintenanceTableClass -> new MFXTableRowCell<>(MaintenanceTableClass::getVehicleModel));
        maintenanceCheck.setRowCellFactory(maintenanceTableClass -> new MFXTableRowCell<>(MaintenanceTableClass::getMaintenanceCheck));
        returnButton.setRowCellFactory(maintenanceTableClass -> {
            MFXButton button = new MFXButton("Return");
            button.setOnAction(actionEvent -> {
            });
            button.setId("tableButton");
            MFXTableRowCell<MaintenanceTableClass, MFXButton> cell = new MFXTableRowCell<>(value -> button, value -> "");
            cell.setGraphic(button);
            return cell;
        });

        // Add the columns and filters to the table
        maintenanceTable.setTableRowFactory(resource -> new MFXTableRow<>(maintenanceTable, resource) {{
            setPrefHeight(50);
        }});
        maintenanceTable.getTableColumns().addAll(plateNumber, vehicleType, maintenanceType, vehicleModelTbl, maintenanceCheck, returnButton);

        // Set the items and autosize the columns
        maintenanceTable.setFooterVisible(false);
        maintenanceTable.autosizeColumnsOnInitialization();
    }`

**Comment by infinite-dev22** [comment]
@Newbie-12131, One solution to such a problem would be using the ID of the data in that row. I suppose the dta populated into your table is gotten from an Object. If so, then you set the Object data with an id property which can later be used to retrieve a specific object, works best with a database. Hope this helps.
